### PR TITLE
refactor: update documentation

### PR DESCRIPTION
**Pull request by whosyourtaco**
## Description

Did some proof-reading of the existing documentation and updated it accordingly, in order for it to be more concise and easily readable.

Also updated a section of the Authentication module that wasn't required.

### CODE REVIEW

**Comment by Christos-Zacharopoulos** [comment]
🌮

File: documentation/docs/modules/Authentication/authentication.md [hunk 1]
```diff
@@ -103,34 +101,18 @@ since the naming is really self-explanatory.
...
+## Appendix
 
-Let's see in depth with some examples the usage of `getAccessTokenSilently`. This is the most interesting one and the one you **WILL** and **MUST** use.
+### Refresh Tokens
 
-As a default set up we use [refresh tokens](https://auth0.com/learn/refresh-tokens/). This has not been said before because it doesn't change the way you use the app.
+Let's deep-dive into the `getAccessTokenSilently` function and how it used under the hood. This is the most important function and the one that you **WILL** and **MUST** use.
 
-One interesting thing though is that, at some point you might make an API call to your BE using the token provided , and the request will fail by being unauthorized.
+By default, we use [refresh tokens](https://auth0.com/learn/refresh-tokens/). This doesn't change the way you use the app, so it wasn't mentioned before.
```
Comment (panvourtsis) [inline documentation/docs/modules/Authentication/authentication.md:112]: i would not mention Auth0 at all as this is a wrapped functionality and auth0 could be replaced in the future and this should still work (open/close principle)

File: documentation/docs/modules/Authentication/authentication.md [hunk 2]
```diff
@@ -103,34 +101,18 @@ since the naming is really self-explanatory.
...
-            cookies.set(__TOKEN__, token);
-          })
-          .catch((error) => {
-            loginWithRedirect();
-        });
-    }
-})
-...
-```
+The toolbox handles all of this logic <b>under the hood</b>, meaning that it's not required to use this function in a specific product.
```
Comment (panvourtsis) [inline documentation/docs/modules/Authentication/authentication.md:118]: we can move this first in this section to let the user know this is handled under the hood for EVERY REQUEST made by using Request tool

File: documentation/docs/modules/Authentication/authentication.md [hunk 3]
```diff
@@ -103,34 +101,18 @@ since the naming is really self-explanatory.
...
+Let's deep-dive into the `getAccessTokenSilently` function and how it used under the hood. This is the most important function and the one that you **WILL** and **MUST** use.
 
-One interesting thing though is that, at some point you might make an API call to your BE using the token provided , and the request will fail by being unauthorized.
+By default, we use [refresh tokens](https://auth0.com/learn/refresh-tokens/). This doesn't change the way you use the app, so it wasn't mentioned before.
 
-This is because the access token has expired. At this point you need to use the function `getAccessTokenSilently` to get a new one using your refresh token.
+One interesting thing about it though, is that at some point you might make an API call to your BE using the token provided and the request will fail as unauthorized.
 
-If this also fails then you need to `loginWithRedirect`.
+This is because the access token has expired. That's when you need to call the `getAccessTokenSilently` function to get a new one, by using your refresh token.
```
Comment (panvourtsis) [inline documentation/docs/modules/Authentication/authentication.md:116]: You don't have to call this actually, if you get 401 or unauthorized by our SSO you will be forced logout

**Review by panvourtsis — APPROVED** [review]

**Pull request merged**